### PR TITLE
fix:  "[NGRM] - Axes2DLayer; Introduce a property "labelFontSizePt" to enable controlling the size of the labels"

### PR DESCRIPTION
**Pull request by nilscb**

### CODE REVIEW

File: typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts [hunk 1]
```diff
@@ -62,15 +62,13 @@ const zDepthAxesAxis = zDepthAxesBackground + 999;
...
 
 const tickLineLength = 10;
-const pixelScale = 8;
 
 export interface Axes2DLayerProps extends ExtendedLayerProps {
     marginH: number;
     marginV: number;
     formatLabelFunc?: (x: number) => string;
     labelColor?: Color;
-    labelFontSize?: number;
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts:72]: This is a breaking change. Suggest keeping the old prop and document it as deprecated.

**Review by hkfb — CHANGES_REQUESTED** [review]

**Comment by hkfb** [comment]
I think it would make sense with a separate story just for font settings

**Review by hkfb — APPROVED** [review]

**Review by hkfb — APPROVED** [review]

File: typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts [hunk 2]
```diff
@@ -914,3 +917,9 @@ function GetTicks(
 
     return ticks;
 }
+
+function GetPixelsScale(labelFontSizePt: number): number {
+    // Estimated number of pixels from baseline to top of font.
+    const px = Math.max(0, (8 / 9) * labelFontSizePt);
```
[the remark this answers is not in the document]
Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts:924]: Added comment to explain the origin of the point-to-pixel formula

**Review by t0oF-azpn — APPROVED** [review]

**Pull request merged**